pkg/utils/utils.go

```diff
@@ -141,3 +141,31 @@ func GetClusterID(cd *hivev1.ClusterDeployment) string {
 		return cd.Spec.ClusterName
 	}
 }
+
+// IsRedHatHyperShiftInfrastructure returns whether or not a cluster is part of the Red Hat HyperShift infrastructure
+func IsRedHatHyperShiftInfrastructure(cd *hivev1.ClusterDeployment) bool {
+	// hypershiftClusterTypeLabel is the annotation key for the hypershift cluster type
+	hypershiftClusterTypeLabel := "ext-hypershift.openshift.io/cluster-type"
+
+	// hypershiftInfrastructureClusterTypes are the values of the hypershiftClusterTypeLabel key
+	// that represent critical Red Hat infrastructure clusters
+	hypershiftInfrastructureClusterTypes := []string{"service-cluster", "management-cluster"}
```

Comment:
This is binding PD operator to labels set by OSDFM, which I think we have to do.  Risk of upstream changes that could result in this no longer being set.  Suggest talking w/ OSDFM team to identify if there is a better solution.  Perhaps a PD specific label, even if set by OSDFM, that is clearly not something that can be changed without some impact downstream.

Suggest:
* key = ext-pagerduty.openshift.io/rh-infra
* value = 'true'

Reply:
Your suggestion to have an  `ext-pagerduty.openshift.io/rh-infra` annotation has been implemented in OSDFM as part of https://issues.redhat.com/browse/OCM-3176. I updated this PR accordingly, but am putting it on hold and will release once the change is fully propagated on staging.